Mac OS GUI tests fail

This is why https://github.com/bartfeenstra/betty/pull/978 skips these tests on Mac OS.

Output:
```
______ TestWelcomeWindow.test_open_project_with_invalid_file_should_error ______

self = <pytestqt.qtbot.QtBot object at 0x11a1aa290>
callback = <function assert_error.<locals>._wait_assert_error.<locals>._assert_error_modal at 0x11a1bc900>

    def waitUntil(self, callback, *, timeout=5000):
        """
        .. versionadded:: 2.0
    
        Wait in a busy loop, calling the given callback periodically until timeout is reached.
    
        ``callback()`` should raise ``AssertionError`` to indicate that the desired condition
        has not yet been reached, or just return ``None`` when it does. Useful to ``assert`` until
        some condition is satisfied:
    
        .. code-block:: python
    
            def view_updated():
                assert view_model.count() > 10
    
    
            qtbot.waitUntil(view_updated)
    
        Another possibility is for ``callback()`` to return ``True`` when the desired condition
        is met, ``False`` otherwise. Useful specially with ``lambda`` for terser code, but keep
        in mind that the error message in those cases is usually not very useful because it is
        not using an ``assert`` expression.
    
        .. code-block:: python
    
            qtbot.waitUntil(lambda: view_model.count() > 10)
    
        Note that this usage only accepts returning actual ``True`` and ``False`` values,
        so returning an empty list to express "falseness" raises a ``ValueError``.
    
        :param callback: callable that will be called periodically.
        :param timeout: timeout value in ms.
        :raises ValueError: if the return value from the callback is anything other than ``None``,
            ``True`` or ``False``.
    
        .. note:: This method is also available as ``wait_until`` (pep-8 alias)
        """
        __tracebackhide__ = True
        import time
    
        start = time.time()
    
        def timed_out():
            elapsed = time.time() - start
            elapsed_ms = elapsed * 1000
            return elapsed_ms > timeout
    
        timeout_msg = f"waitUntil timed out in {timeout} milliseconds"
    
        while True:
            try:
>               result = callback()

/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pytestqt/qtbot.py:525: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def _assert_error_modal():
        nonlocal widget
        widget = qapp.activeModalWidget()
>       assert isinstance(widget, error_type)
E       AssertionError: assert False
E        +  where False = isinstance(None, <class 'betty.gui.error.ExceptionError'>)

betty/tests/conftest.py:158: AssertionError

The above exception was the direct cause of the following exception:

self = <betty.tests.gui.test_app.TestWelcomeWindow object at 0x115479310>
assert_error = <function assert_error.<locals>._wait_assert_error at 0x119f82200>
mocker = <pytest_mock.plugin.MockerFixture object at 0x11a1a9d10>
qtbot = <pytestqt.qtbot.QtBot object at 0x11a1aa290>
tmp_path = PosixPath('/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pytest-of-runner/pytest-0/test_open_project_with_invalid0')

    def test_open_project_with_invalid_file_should_error(self, assert_error, mocker: MockerFixture, qtbot: QtBot, tmp_path: Path) -> None:
        with App() as app:
            sut = WelcomeWindow(app)
            qtbot.addWidget(sut)
            sut.show()
    
            configuration_file_path = tmp_path / 'betty.json'
            # Purposefully leave the file empty so it is invalid.
            configuration_file_path.write_text('')
            mocker.patch.object(QFileDialog, 'getOpenFileName', mocker.MagicMock(return_value=[configuration_file_path, None]))
            qtbot.mouseClick(sut.open_project_button, Qt.MouseButton.LeftButton)
    
>           error = assert_error(ExceptionError)

betty/tests/gui/test_app.py:76: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

error_type = <class 'betty.gui.error.ExceptionError'>

Error:     def _wait_assert_error(error_type: Type[Error]) -> Error:
        widget = None
    
        def _assert_error_modal():
            nonlocal widget
            widget = qapp.activeModalWidget()
            assert isinstance(widget, error_type)
>       qtbot.waitUntil(_assert_error_modal)
E       pytestqt.exceptions.TimeoutError: waitUntil timed out in 5000 milliseconds

betty/tests/conftest.py:159: TimeoutError
```

--- a/betty/tests/gui/test_app.py
+++ b/betty/tests/gui/test_app.py
@@ -1,5 +1,4 @@
 import json
-import sys
 from os import path
 from pathlib import Path
 
@@ -74,7 +73,6 @@ class TestBettyMainWindow:
 
 
 class TestWelcomeWindow:
-    @pytest.mark.skipif(sys.platform == 'darwin', reason='See https://github.com/bartfeenstra/betty/issues/982')
     async def test_open_project_with_invalid_file_should_error(
         self,
         assert_error: AssertError[ExceptionError],

